From: Anton Arapov <aarapov@redhat.com>
Date: Wed, 2 Jan 2008 14:46:14 +0100
Subject: [net] get minimum RTO via tcp_rto_min
Message-id: 20080102134614.GA25938@bandura.englab.brq.redhat.com
O-Subject: Re: [RHEL5.2 PATCH] BZ427205: make tcp_input_metrics() get minimum RTO via tcp_rto_min()
Bugzilla: 427205

Details:
  This patch is missed part of BZ303011 patch.
  https://bugzilla.redhat.com/show_bug.cgi?id=303011
  tcp_input_metrics() refers to the built-time constant TCP_RTO_MIN
regardless of
configured minimum RTO with iproute2.

Upstream status:
  in upstream
  http://git.kernel.org/?p=linux/kernel/git/davem/net-2.6.git;a=commit;h=488faa2ae32c2f8b1ffda5c7fae5eb7c979acc15

Test status:
  patch has been tested for compilation and boot.

Acked-by: Arnaldo Carvalho de Melo <acme@redhat.com>
Acked-by: Anton Arapov <aarapov@redhat.com>

diff --git a/net/ipv4/tcp_input.c b/net/ipv4/tcp_input.c
index 684fc4d..54dea85 100644
--- a/net/ipv4/tcp_input.c
+++ b/net/ipv4/tcp_input.c
@@ -840,7 +840,7 @@ static void tcp_init_metrics(struct sock *sk)
 	}
 	if (dst_metric(dst, RTAX_RTTVAR) > tp->mdev) {
 		tp->mdev = dst_metric(dst, RTAX_RTTVAR);
-		tp->mdev_max = tp->rttvar = max(tp->mdev, TCP_RTO_MIN);
+		tp->mdev_max = tp->rttvar = max(tp->mdev, tcp_rto_min(sk));
 	}
 	tcp_set_rto(sk);
 	tcp_bound_rto(sk);
